velox/exec/SortWindowBuild.cpp

```diff
@@ -179,15 +181,37 @@ void SortWindowBuild::computePartitionStartRows() {
   partitionStartRows_.push_back(0);
 
   VELOX_CHECK_GT(sortedRows_.size(), 0);
-  for (auto i = 1; i < sortedRows_.size(); i++) {
-    if (partitionCompare(sortedRows_[i - 1], sortedRows_[i])) {
-      partitionStartRows_.push_back(i);
+  int32_t start = 0;
+  if (binarySearch_) {
```

Comment:
Please add a comment here with the explanation of the algorithm you provided in the PR discussion.

Reply:
done